src/Calculator.ts

```diff
@@ -213,7 +211,63 @@ export function calculateMeasureReports(
         });
       }
 
-      createMeasureScore(measure, group, detail, report);
+      // if the measure definition has stratification, add group.stratifier
+      const stratifier = measure.group?.find(g => g.id == detail.groupId)?.stratifier;
+      if (stratifier) {
+        group.stratifier = [];
+        stratifier.forEach(s => {
+          const reportStratifier = <R4.IMeasureReport_Stratifier>{};
+          reportStratifier.code = s.code ? [s.code] : [];
+          const strat = <R4.IMeasureReport_Stratum>{};
+          // use existing populations, but reduce count as appropriate
+          strat.population = JSON.parse(JSON.stringify(group.population));
+
+          if (detail.episodeResults) {
+            detail?.episodeResults?.forEach(er => {
+              const inStrat = er.stratifierResults?.find(sr => sr.strataCode == s.code?.text)?.result;
+              strat.population?.forEach(pop => {
+                const result = er.populationResults.find(pr => {
+                  return (
+                    pop.code?.coding && pop.code.coding.length > 0 && pop.code.coding[0].code === pr.populationType
+                  );
+                })?.result;
+                if (result && !inStrat && pop.count) {
+                  // reduce count if this episode is in this population but not the stratification
+                  pop.count = pop.count - 1;
+                }
+              });
+            });
+          } else {
+            const inStrat = detail.stratifierResults?.find(sr => sr.strataCode === s.code?.text)?.result;
+            // start with population count and change count to 0 if not in strattification
+            strat.population?.forEach(pop => {
+              if (!inStrat) pop.count = 0;
+            });
+          }
+
+          const scoringCode =
+            measure.scoring?.coding?.find(c => c.system === 'http://hl7.org/fhir/measure-scoring')?.code || '';
+          if (scoringCode === MeasureScoreType.CV) {
+            // TODO: should we add score observation for stratification?
+            strat.measureScore = calcMeasureScoreCV(measure, detail, group.id || '', s);
+          } else {
+            strat.measureScore = calcMeasureScore(scoringCode, JSON.parse(JSON.stringify(strat.population)));
```

Comment:
Q: what's the need for the parse/stringify combo here?

Reply:
Deep copy + matching type structures that aren't actually the same type